vllm/config.py

```diff
@@ -3240,14 +3243,20 @@ def compute_hash(self) -> str:
                                usedforsecurity=False).hexdigest()
         return hash_str
 
+    @classmethod
+    def from_dict(cls,
+                  dict_value: dict[str, Any] | None) -> StructuredOutputConfig:
+        """Parse the CLI value for the speculative config."""
+        return cls(**(dict_value or {}))
```

Comment:
`from_dict` isn't necessary:
- Not all configs have it, so it introduces inconsistency
- It's trivial to construct a dataclass from a dict using `StructuredOutput(**the_dict)`